### DISCUSSION TITLE
3D Cruise to use GPS speed instead of using throttle setting

### DISCUSSION THREAD

**Discussion by quetiano** [iNavFlight/inav#8918] [Ideas]
I thought that this would be a something beneficial for range flyers as well as myself in those windy conditions when doing a range flight  into headwind.
Since the 3D cruise mode is set using a throttle setting and if you are in a various gusting headwinds, could lead to a stall while your plane is many km's/miles away.
I thought of the idea...what if 3D Cruise can use gps speed as to maintain a minimum speed, a minimum speed which you can set in the Inav configurator... and when flying in 3D cruise mode, Inav will never let the plane fall below that set speed, in any weather condition ?
For example, a calm day exists here in South Africa, like yeah, that would be a cold day in hell...but, lets imagine its calm... and I set a minimum speed of 60 km/h, so when flying in 3D cruise, Inav will maintain 60 km/h by continuously adjusting throttle to keep gps speed at 60 km/h...suddenly, at 8 km out, a headwind begins at 50 km/h, then Inav will quickly increase throttle to keep the gps speed at 60 km/h, and when turning around to return, Inav will decrease throttle to keep that 60 km/h.

Is this possible to change 3D cruise or create this feature in Inav ?

**Comment by Jetrell** [comment]
@quetiano GPS speed isn't an ideal reference to use.. 
Because its not efficient when the plane is traveling into a headwind...  And it will risk a stall, when traveling with a tail wind.

**e.g.** If you have a very strong headwind. The FC will apply as much throttle as is required, to maintain the speed set point, and waste energy in the process.
But if the plane has a strong tailwind. The FC will reduce the throttle, to stop the plane exceeding the speed set point... And if it encounters a micro-burst or down-draft, that require elevator input to hold altitude, it will risk a stall.

Air speed is a better reference for efficiency and flight safety.

**Reply by quetiano** [reply]
Thank you for your awesome reply Jetrell !
In regards to my worry about stalling when flying a 5+ km trip, in 3D cruise mode, any ideas on what can I do to keep the plane flying without stalling while without really knowing what the headwind really is ?

**Reply by Jetrell** [reply]
> Thank you for your awesome reply Jetrell ! In regards to my worry about stalling when flying a 5+ km trip, in 3D cruise mode, any ideas on what can I do to keep the plane flying without stalling while without really knowing what the headwind really is ?

The current throttle control method INAV uses, is safe enough..
A stall is only likely to occur if your throttle values are set too conservatively, for the wind conditions you may encounter. Or if your navigation climb angle is set too high. (If your plane is heavy. No more than 22°)
You just have to make sure your cruise throttle settings are optimized to suit the requirement of your fixed wings thrust to weight.

![FW nav settings](https://user-images.githubusercontent.com/47995726/228381249-3dd11ad0-8d14-4e6d-981c-b0930087434a.JPG)

I use Cruse mode all the time on my fixed wings. And travel great distances. Often in windy conditions. And I've never had a plane stall.

**Reply by Machatron** [reply]
> The current throttle control method INAV uses, is safe enough..

No it's not.
> A stall is only likely to occur if your throttle values are set too conservatively, for the wind conditions you may encounter.

Is that the user's fault? Yesterday inav on autopilot almost stalled my plane multiple times and i had to take over. All of these occurrences happened when it was flying in a headwind and the 50% throttle the autopilot was using was insufficient. My 3d speed was dropping below 30mph (danger zone on my craft). When turned around flying in a tailwind it was flying much faster at 50mph+ with the same 50% cruise throttle. today i went to fly again but cranked up my cruise throttle to 65% to fight the wind better. But today by luck wasn't windy at all. So the result was my plane (on navigation modes) was flying the whole time much faster than it needed and wasting energy. So depending on which direction i was flying (headwind or tailwind) neither config would be sufficient. Either with 50% cruise throttle it flies fine in a tailwind but stalls in a headwind, or with 65% cruise throttle it flies fine in a headwind and fly's excessively fast and inefficiently in a tailwind.

Is it the users responsibility to constantly update their cruise configs before each flight? what if wind conditions change during flight? Yea sure i can have different profiles on a switch and change profiles with different cruise throttles depending on wind conditions but wouldn't it be better if all this was automatic?

I think this discussion was shut down too hastily. I would love for a controller to enforce a minimum air/3d speed that would take priority above the standard cruise throttle + pitch to throttle controller. This would only need to be a single config parameter in the "fixed wing navigation settings" of the configurator. maybe between "cruise throttle" and "allow manual throttle increases" with the unit being in cm/s, default value around 1700, and the name being "minimum air speed".

The current way cruise speed is controlled is very blind because it isn't controlled at all. The autopilot is just blindly cruising at a preset throttle and linearly adjusting it depending on pitch. There is no account for external conditions like headwinds even when sensor data from gps can provide early indicators the craft is about the stall and it needs more throttle.

Minimum 3d gps speed would be a very cool addition to inav and i would love for this to be added.

**Comment by MrD-RC** [comment]
> > The current throttle control method INAV uses, is safe enough..
> 
> No it's not.

Yes it is

> > A stall is only likely to occur if your throttle values are set too conservatively, for the wind conditions you may encounter.
> 
> Is that the user's fault? Yesterday inav on autopilot almost stalled my plane multiple times and i had to take over. All of these occurrences happened when it was flying in a headwind and the 50% throttle the autopilot was using was insufficient. 

When cruise throttle is used, the aircraft doesn’t care if it’s a headwind or any other kind of wind. The wind direction is relative to the ground, not the aircraft. The only time it matters is for takeoff, landing, and energy calculations. If your aircraft was on the verge of stalling, the cruise throttle is too low for your aircraft. Period. As a note, you can fly backwards with a strong enough headwind. All the aircraft sees is the airspeed relative to it.

> My 3d speed was dropping below 30mph (danger zone on my craft). When turned around flying in a tailwind it was flying much faster at 50mph+ with the same 50% cruise throttle. 

3D Speed is based on ground speed and vertical speed. It doesn’t take the wind into consideration. If in to wind you’re doing 30 and, with the same throttle setting, with the wind you’re doing 50. There is a 10mph wind. Meaning the whole time, your plane is seeing 40mph. So if 30mph is the danger zone for your craft, you had 10mph of buffer. In fact; if you had flown with the wind at 30mph, the airplane would only see 20mph, which would likely stall.

> Today i went to fly again but cranked up my cruise throttle to 65% to fight the wind better. But today by luck wasn't windy at all. So the result was my plane (on navigation modes) was flying the whole time much faster than it needed and wasting energy. So depending on which direction i was flying (headwind or tailwind) neither config would be sufficient. Either with 50% cruise throttle it flies fine in a tailwind but stalls in a headwind, or with 65% cruise throttle it flies fine in a headwind and fly's excessively fast and inefficiently in a tailwind.
> 

Again, you’re not understanding how the wind works. Or even how range works. For the same throttle, the airplane won’t be working any harder or easier depending on the wind or wind direction. All that will be different is the speed relative to the ground. Also, flying with the minimum possible throttle to prevent stall does not equal range, only time. More range can be achieved by flying a little faster. It uses a bit more energy, but covers more ground. It’s called the Carson speed. 

> Is it the users responsibility to constantly update their cruise configs before each flight? what if wind conditions change during flight? Yea sure i can have different profiles on a switch and change profiles with different cruise throttles depending on wind conditions but wouldn't it be better if all this was automatic?

Yes, it is the pilots responsibility to adjust their flight parameters. The same as needing to change the RTH method depending on the flying environment. Set cruise throttle for an average day. This will ensure it won’t stall. Then you can use manual throttle override to add more throttle if it isn’t covering ground quickly enough.

> I think this discussion was shut down too hastily. I would love for a controller to enforce a minimum air/3d speed that would take priority above the standard cruise throttle + pitch to throttle controller. This would only need to be a single config parameter in the "fixed wing navigation settings" of the configurator. maybe between "cruise throttle" and "allow manual throttle increases" with the unit being in cm/s, default value around 1700, and the name being "minimum air speed".
> 
> The current way cruise speed is controlled is very blind because it isn't controlled at all. The autopilot is just blindly cruising at a preset throttle and linearly adjusting it depending on pitch. There is no account for external conditions like headwinds even when sensor data from gps can provide early indicators the craft is about the stall and it needs more throttle.
> 
> Minimum 3d gps speed would be a very cool addition to inav and i would love for this to be added.

No, the current method isn’t blind at all. Once you understand that the aircraft, therefore throttle, doesn’t care about ground speed, or anything derived from it; it makes perfect sense. Headwinds and tailwinds don’t matter. All that matters, with regards to not entering a stall when cruising, is airspeed. 

BTW, stalls are really not about that either. They are based on the attachment or air to the wing, and can potentially happen at any air speed; depending on the wing profile and AoA. I’ve just be talking very basic about an aircraft flying straight and level, too slowly. 

There is a minimum ground speed built in to INAV. I believe it’s 7m/s. Which should stop most aircraft from entering stall. Maybe making this setting accessible would be good. But the problem is it could be misused.
